blueocean-dashboard/src/main/js/creation/bitbucket/cloud/BbCloudFlowManager.js

```diff
@@ -0,0 +1,414 @@
+    _updateRepositories(organizationName, repoData) {
+        const { items, nextPage } = repoData.repositories;
+        const firstPage = this.repositories.length === 0;
+        const morePages = !isNaN(parseInt(nextPage, 10));
+
+        this.repositories.push(...items);
+        this._repositoryCache[organizationName] = this.repositories.slice();
+
+        if (morePages) {
+            this._loadPagedRepository(organizationName, nextPage)
+                .then(repos2 => this._updateRepositories(organizationName, repos2, nextPage));
+        } else {
+            this.repositoriesLoading = false;
+        }
+        if (firstPage) {
+            // render the repo list only once, after the first page comes back
+            // otherwise we'll lose step's internal state
+            this.renderStep({
+                stateId: STATE.STEP_CHOOSE_REPOSITORY,
+                stepElement: <BbRepositoryStep />,
+                afterStateId: STATE.STEP_CHOOSE_ORGANIZATION,
+            });
+        }
+    }
+
+    saveRepo() {
+        this._saveRepo();
+    }
+
+    @action
+    _saveRepo() {
+        const afterStateId = this.isStateAdded(STATE.STEP_RENAME) ?
+            STATE.STEP_RENAME : STATE.STEP_CHOOSE_REPOSITORY;
+
+        this.renderStep({
+            stateId: STATE.STEP_COMPLETE_SUCCESS,
+            stepElement: <BbCloudCompleteStep />,
+            afterStateId,
+        });
+
+        this.setPlaceholders();
+
+        this._initListeners();
+
+        this._creationApi.createMbp(this.credentialId, this.getApiUrl(), this.pipelineName, this.selectedOrganization.key, this.selectedRepository.name)
+            .then(waitAtLeast(MIN_DELAY * 2))
+            .then(result => this._createPipelineComplete(result));
+    }
+
+    saveRenamedPipeline(pipelineName) {
+        this.pipelineName = pipelineName;
+        return this._saveRepo();
+    }
+
+    @action
+    _createPipelineComplete(result) {
+        this.outcome = result.outcome;
+        if (result.outcome === CreateMbpOutcome.SUCCESS) {
+            this.changeState(STATE.STEP_COMPLETE_SUCCESS);
+            this.pipeline = result.pipeline;
+        } else if (result.outcome === CreateMbpOutcome.INVALID_NAME) {
+            this.renderStep({
+                stateId: STATE.STEP_RENAME,
+                stepElement: <BbRenameStep pipelineName={this.pipelineName} />,
+                afterStateId: STATE.STEP_CHOOSE_REPOSITORY,
+            });
+            this._showPlaceholder();
+        } else if (result.outcome === CreateMbpOutcome.INVALID_URI || result.outcome === CreateMbpOutcome.INVALID_CREDENTIAL) {
+            this.removeSteps({ afterStateId: STATE.STEP_CREDENTIAL });
+            this._showPlaceholder();
+        } else if (result.outcome === CreateMbpOutcome.ERROR) {
+            this.renderStep({
+                stateId: STATE.ERROR,
+                stepElement: <BbUnknownErrorStep error={result.error} />,
+                afterStateId: STATE.STEP_CONNECT,
```

Comment:
This looks like a copy/paste error from Git creation. You probably need to check if `STATE.STEP_NAME` is added and show it after either that or `STATE.STEP_CHOOSE_REPOSITORY`